### PR TITLE
create a wafflejs helper to include without HTTP request

### PR DESCRIPTION
**Pull request by groovecoder**
Found this to be a helpful approach to avoid the extra HTTP request for waffle flags in JavaScript.

### CODE REVIEW

**Comment by groovecoder** [comment]
Pull #100 !! w00t!

File: waffle/helpers.py [hunk 1]
```diff
@@ -9,8 +10,14 @@ def flag_helper(context, flag_name):
     return flag_is_active(context['request'], flag_name)
 
 
+@jinja2.contextfunction
+def wafflejs_helper(context):
```
Comment (jsocol) [inline waffle/helpers.py]: This should have the word `inline` in its name somewhere

File: docs/usage.rst [hunk 2]
```diff
@@ -311,6 +320,16 @@ Waffle JavaScript in your templates::
 
     <script src="{% url wafflejs %}"></script>
 
+wafflejs helper
```
Comment (jsocol) [inline docs/usage.rst]: Waffle has always supported both Jinja and Django templates. How much work would it be to get the helper to work for both?

File: waffle/views.py [hunk 3]
```diff
@@ -11,6 +11,10 @@
 
 @never_cache
 def wafflejs(request):
+    return _generate_waffle_js_resp(request)
+
+
+def _generate_waffle_js_resp(request):
```
Comment (jsocol) [inline waffle/views.py]: Is it actually producing an `HttpResponse`, and if so, maybe it's better that the intermediate function return a string and then have the view wrap it in a response object, and the template helper just dump out the string.

File: docs/usage.rst (hunk 2 above)
Comment (groovecoder) [inline docs/usage.rst]: This works in Jinja for sure - that's how I'm using it.

**Comment by jsocol** [comment]
#100! :wine_glass: I could've sworn there was a champagne glass emoji

File: waffle/views.py (hunk 3 above)
Comment (groovecoder) [inline waffle/views.py]: Fixed.

File: waffle/helpers.py (hunk 1 above)
Comment (groovecoder) [inline waffle/helpers.py]: Fixed.

**Comment by jsocol** [comment]
Mind adding some kind of test for both jinja and django templates, just to prove it renders?

**Comment by groovecoder** [comment]
Think I got everything. How's this look for a merge? Would love to stay on the `jsocol` remote for this submodule in kuma-lib. :)

**Comment by jsocol** [comment]
With two new passing tests, I will eagerly hit the blue button :)

**Comment by groovecoder** [comment]
What should I test?

**Comment by jsocol** [comment]
Add a template that uses the helper (within `<script>` tags or not) and just assert that `window.waffle` shows up in the rendered response? See the other template helper tests for a rough outline. Will probably take separate templates for jingo and django.

**Comment by groovecoder** [comment]
There are template tag tests, but no other template helper tests.

**Comment by jsocol** [comment]
[`waffle/tests/test_templates.py`](https://github.com/jsocol/django-waffle/blob/master/waffle/tests/test_templates.py#L21) tests both the Django template tags and the Jingo helpers.

My point here is that this does not support Django templates right now. We can do that with a new [inclusion tag](https://docs.djangoproject.com/en/1.5/howto/custom-template-tags/#inclusion-tags) in `waffle/templatetags/waffle_tags.py`. Tests would demonstrate that it fails. I will merge this as soon as there are two new asserts (preferably new tests) that demonstrate the jingo render path and the django render path actually outputting text.

These can probably be added to the existing test_app templates. For the django inclusion tag, something like `{% waffle_js %}` would be perfect.

**Comment by groovecoder** [comment]
Updated with tests!
tests

**Comment by jsocol** [comment]
Woo, thanks! :beers:

**Pull request merged**